CompilerSource/parsing/parser.cpp

```diff
@@ -383,15 +383,12 @@ bool AstBuilder::next_maybe_functional_cast() {
 }
 
 std::unique_ptr<AST::DeclarationStatement> AstBuilder::parse_declarations(
-    AST::DeclarationStatement::StorageClass sc, jdi::definition *def,
-    AST::DeclaratorType decl_type,
-  bool parse_unbounded, std::vector<AST::DeclarationStatement::Declaration> decls, bool already_parsed_first ) {
+    AST::DeclarationStatement::StorageClass sc, FullType &ft, AST::DeclaratorType decl_type, bool parse_unbounded,
```

Comment:
I modified this function to accept the entire `FullType`, not just the `definition`, because we also need the flags. The original function assigned the same default flag (`signed`) to all declarations, as it didn't account for the parsed flags.